src/modules/Host.rb

```diff
@@ -207,14 +207,15 @@ def Import(settings)
     # Dump the Hosts settings to a map, for autoinstallation use.
     # @return autoinstallation settings
     def Export
-      if Ops.greater_than(Builtins.size(@hosts), 0)
-        # Filter out IPs with empty hostname (so that valid autoyast
-        # profile is created)(#335120)
-        @hosts = Builtins.filter(@hosts) { |_ip, names| names != [] }
-        return { "hosts" => Builtins.eval(@hosts) }
-      else
-        return {}
-      end
+      return {} if hosts.empty?
+
+      # Filter out IPs with empty hostname (so that valid autoyast
+      # profile is created)(#335120)
+      # FIXME: this also removes records with empty names from @hosts. Such
+      # side effect is unexpected and should be removed.
```

Comment:
A link to the original bug, in which  alternative approaches are discussed: https://bugzilla.suse.com/show_bug.cgi?id=335120